public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AddAnomalyDetector.tsx

```diff
@@ -0,0 +1,823 @@
+                value.featureId,
+                value.aggregationBy,
+                value.aggregationOf[0].label
+              ),
+        newFeature: true,
+        aggregationOf: formikToAggregationOf(value),
+      };
+
+      return oneFeature;
+    });
+  }
+
+  const initialDetectorValue = {
+    name: detectorNameFromVis,
+    indices: [embeddable.vis.data.aggs.indexPattern.title],
+    timeField: embeddable.vis.data.indexPattern.timeFieldName,
+    detectionInterval: {
+      period: { interval: intervalValue, unit: UNITS.MINUTES },
+    },
+    windowDelay: {
+      period: { interval: delayValue, unit: UNITS.MINUTES },
+    },
+    shingleSize: shingleSizeValue,
+    filterQuery: { match_all: {} },
+    description: '',
+    resultIndex: undefined,
+    filters: [],
+    featureList: featureListToFormik(featureList),
+    categoryFieldEnabled: false,
+    realTime: true,
+    historical: false,
+  };
+
+  const handleAssociate = async (detectorId: string) => {
+    console.log('inside handleAssociate');
+    enum VisLayerTypes {
+      PointInTimeEvents = 'PointInTimeEvents',
+    }
+
+    const fn = {
+      type: VisLayerTypes.PointInTimeEvents,
+      name: 'overlay_anomalies',
+      args: {
+        detectorId: detectorId,
+      },
+    } as VisLayerExpressionFn;
+
+    const savedObjectToCreate = {
+      title: 'test-title',
+      pluginResourceId: detectorId,
+      visId: embeddable.vis.id,
+      visLayerExpressionFn: fn,
+    } as ISavedAugmentVis;
+
+    const savedObject = await createAugmentVisSavedObject(savedObjectToCreate);
+
+    const response = await savedObject.save({});
+    closeFlyout();
+  };
+
+  function formikToDetectorName(title) {
+    const detectorName =
+      title +
+      '_anomaly_detector_' +
+      Math.floor(100000 + Math.random() * 900000);
+    detectorName.replace(/[^a-zA-Z0-9-_]/g, '_');
+    return detectorName;
+  }
+
+  function formikToAggregationOf(value) {
+    return [
+      {
+        label: value.aggregationOf[0].label,
+        type: 'number',
+      },
+    ];
+  }
+
+  function formikToUIMetadataAggregation(featureId, aggMethod, fieldName) {
+    const returnVal = {
+      [snakeCase(getFeatureNameFromParams(featureId))]: {
+        [aggMethod]: { field: fieldName },
+      },
+    };
+    return returnVal;
+  }
+
+  function formikToAggregation(value) {
+    return {
+      [snakeCase(getFeatureNameFromParams(value.id))]: {
+        avg: { field: value.params.field.name },
+      },
+    };
+  }
+
+  function featureListToFormik(featureList): FeaturesFormikValues[] {
+    return featureList.map((feature) => {
+      return {
+        featureId: feature.id,
+        featureName: getFeatureNameFromParams(feature.id),
+        featureEnabled: true,
+        featureType: FEATURE_TYPE.SIMPLE,
+        aggregationBy: 'avg',
+        aggregationOf: [{ label: feature.params.field.name }],
+        aggregationQuery: formikToAggregation(feature),
+      };
+    });
+  }
```

Comment:
These should all exist in some form already in helper files. For example, https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/ReviewAndCreate/utils/helpers.ts

Reply:
I checked the existing helper file while doing this part and tried to reuse the existing functions as much as possible, however, since we do prefill from the vis, so we need some function to convert vis information to feature list. Currently in the existing create flow, this part is handled by the create detector logic - we parse the formik inputs to feature list. So when we display it later, it's in the correct feature list format. But for feature anywhere, we need some new helper function to convert vis feature list information to correct feature list format.

Reply:
I'm still a little confused. If you structure the formik fields to be formatted the same as the existing creation formik fields, then that should eliminate needing any formik conversion functions, or needing new components (can reuse old components if the same formik fields are passed to it.

Understood there may be some edge cases to convert some vis values _to_ formik, but that should just be a one time operation, and then everything downstream can use existing logic/components. For this part, I think it would be best to have a single `visToFormik()` fn or something to populate all fields that need to be specially set based on the vis config.